File: 04_Prime Number/prime_number.c
#include <stdio.h>

int main()
{

    int num, i, j, r1, r2, flag;

    // Program for checking number is prime or not

    printf("Enter any number: ");
    scanf("%d", &num);

    if (num != 1)
    {

        for (i = 2; i <= num; i++)
        {
            if (num % i == 0)
            {
                break;
            }
        }
        if (i == num)
        {
            printf("Number is prime \n");
        }
        else
        {
            printf("Not a prime number. \n");
        }
    }
    else
    {
        printf("Not a prime number \n");
    }

    // Program for printing prime number in a given range.

    printf("Enter lower bound: ");
    scanf("%d", &r1);

    printf("Enter upper bound: ");
    scanf("%d", &r2);

    if (r2 < r1)
    {
        printf("Invalid range!");
    }
    else
    {
        for (i = r1; i <= r2; i++)
        {

            flag = 1;
            if (i != 1)
            {
                for (j = 2; j < i; j++)
                {

                    if (i % j == 0)
                    {
                        flag = 0;
                        break;
                    }
                }

                if (flag == 1)
                {
                    printf("%d", i);
                }
            }
        }
    }

    return 0;
}

File: 04_Prime Number/prime_number.py

# Program for checking prime numbers.

num = input("Enter any number: ")

num = int(num)

if(num != 1):
    for i in range(2, num):
        if(num % i) == 0:
            print(num, "is not a prime number")
            break
    else:
        print(num, "is a prime number")
else:
    print(num, "Not a prime number.")


# Program for printing n prime numbers in a given range.

r1,r2 = int(input("Lower bound: ")),int(input("Upper bound: "))

if(r2 < r1):
    print("Invalid Range")
    exit()

print("Prime numbers are: ", sep=" ", end="")

for i in range(r1, r2+1):
    if i > 1:
        for j in range(2, i):

            if(i%j == 0):
                break
        else:
            print(i, sep=", ", end=" ")
